### PR TITLE
CMake: Build unittest from mbed-os root directory

### PR DESCRIPTION
**Pull request by rajkan01**
### Summary of changes 
- Added new CMake command line macro CMAKE_UNIT_TESTING in mbed-os/CMakeLists.txt file to add unittest directory which will allow us to build the unittest from mbed os root directory 
- Used fetch content module to fetch google test instead of using .in config file

Note: This is breaking change:
- CMake issue warnings as I have moved all below CMake config from UNITTESTS/CMakeLists.txt into mbed-os/CMakeLists.txt
```
    set(PROJECT_NAME unittests)
    set(LIB_NAME MbedOS)
    project(${PROJECT_NAME})
```
- I have modified the `unittest-includes-base` to add `UNITTESTS` directory as we try to run unittest from mbed os root directory, so Mbed CLI 1 fails with missing headers when we run a unit test using `mbed test --unittests command`

### Documentation 

----------------------------------------------------------------------------------------------------------------
### Pull request type 

    [] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results 

    [] No Tests required for this change (E.g docs only update)
    [] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers 
@hugueskamba @0xc170

----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,6 +5,12 @@
 
 cmake_minimum_required(VERSION 3.19.0 FATAL_ERROR)
 
+if(${CMAKE_UNIT_TESTING})
+    set(PROJECT_NAME unittests)
+    set(LIB_NAME MbedOS)
+    project(${PROJECT_NAME})
+    add_subdirectory(UNITTESTS)
```
Comment (0xc0170) [inline CMakeLists.txt:11]: shouldn't be here jsut subdirectory and the other 3 variables to be set in the unittests CMakelists?

File: UNITTESTS/CMakeLists.txt [hunk 2]
```diff
@@ -102,62 +67,62 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DUNITTEST")
...
-  "${PROJECT_SOURCE_DIR}/../connectivity/cellular/include/cellular/framework/common"
-  "${PROJECT_SOURCE_DIR}/../connectivity"
-  "${PROJECT_SOURCE_DIR}/../connectivity/lorawan/include/lorawan"
-  "${PROJECT_SOURCE_DIR}/../connectivity/lorawan/lorastack"
-  "${PROJECT_SOURCE_DIR}/../connectivity/lorawan/lorastack/mac"
-  "${PROJECT_SOURCE_DIR}/../connectivity/lorawan/lorastack/phy"
-  "${PROJECT_SOURCE_DIR}/../connectivity/lorawan"
-  "${PROJECT_SOURCE_DIR}/../connectivity/mbedtls"
-  "${PROJECT_SOURCE_DIR}/../connectivity/mbedtls/include"
+  "${PROJECT_SOURCE_DIR}/UNITTESTS/target_h"
```
Comment (0xc0170) [inline UNITTESTS/CMakeLists.txt]: wouldn't this work with relative paths as we use in the Mbed OS as well?

`./target_h` ?

File: tools/cmake/README.md [hunk 3]
```diff
@@ -96,7 +96,20 @@ $ mbedtools configure -t <TOOLCHAIN> -m <MBED_TARGET>
...
   ```
-
 Notes:
 * These steps will change when `mbedtools` implements a sub-command to invoke Greentea tests
 * Some Greentea tests require specific application configuration files in order to build and run successfully. For example, the `connectivity/mbedtls/tests/TESTS/mbedtls/sanity` test requires the configuration file found at `TESTs/configs/experimental.json`.
+
+## How to build and run a unit test
+
+Install prerequisites suggested in the previous section and follow the below steps:
+* Run the following command to build the unit test from mbed-os root directory:
```
Comment (0xc0170) [inline tools/cmake/README.md:106]: You mentioned it does not work from the root - but it states here it does?

File: CMakeLists.txt (hunk 1 above)
Comment (rajkan01) [reply CMakeLists.txt:11]: Cmake will issue a warning
```
CMake Warning (dev) in CMakeLists.txt:
  No project() command is present.  The top-level CMakeLists.txt file must
  contain a literal, direct call to the project() command.  Add a line of
  code such as

    project(ProjectName)

  near the top of the file, but after cmake_minimum_required().

  CMake is pretending there is a "project(Project)" command on the first
  line.

```

Comment (0xc0170) [reply CMakeLists.txt:11]: This is similar to tests, unittests should be the top level CMakefile, and MBed OS should be included there. The same as we do the applications. unittests are basically apps.

This should be removed then and main cmakefile

File: tools/cmake/README.md (hunk 3 above)
Comment (rajkan01) [reply tools/cmake/README.md:106]: With these PR changes, we can build unittest but can't run from the mbed-os root directory.

**Pull request closed without merging**